Frontend/src/app/components/user-directory/user-directory.component.html

```diff
@@ -1,39 +1,77 @@
 <h2>User Directory</h2>
-<p>
-  Displaying
-  {{ loadingService.isLoadingId(loadingKey) ? "?-? of ?" : paginationText }}
-  users.
-</p>
-<app-pagination-controls
-  [(page)]="paginationSignals.page.signal"
-  [(pageSize)]="paginationSignals.pageSize.signal"
-  [loadingId]="loadingKey"
-  [paginatedResult]="paginatedResultSignal()"
-/>
+
 @if (
   !loadingService.isLoadingId(loadingKey) &&
   !paginatedResultSignal()?.pagination
 ) {
   <p>Failed to load pagination data.</p>
 }
-<form [formGroup]="searchForm" (ngSubmit)="onSearchSubmit()" autocomplete="off">
-  <app-text-input
-    label="Username"
-    formControlName="username"
-    [errorMessages]="{
-      maxlength: 'Usernames are at most 20 characters long.',
-    }"
-    [showErrors]="submitted()"
-  />
-  <button>Search</button>
-  <button type="button" (click)="onResetSearch()">Reset</button>
+<form
+  [formGroup]="searchForm"
+  (ngSubmit)="onSearchSubmit()"
+  autocomplete="off"
+  class="form-container"
+>
+  <div class="search-bar">
+    <app-text-input
+      label="Username"
+      formControlName="username"
+      [showLabel]="false"
+      [errorMessages]="{
+        maxlength: 'Usernames are at most 20 characters long.',
+      }"
+      [showErrors]="submitted()"
+    />
+    <button class="btn">
+      Search
+      <ng-icon name="matSearch" />
+    </button>
+    <button class="btn" type="button" (click)="onResetSearch()">
+      Reset
+      <ng-icon name="matRestartAlt" />
+    </button>
+  </div>
 </form>
-@if (items()[0]) {
-  <ul>
-    @for (user of items(); track user.id) {
-      <li>{{ user.id }}, {{ user.username }}</li>
-    }
-  </ul>
-} @else if (!loadingService.isLoadingId(loadingKey) && !searchForm.pristine) {
-  <p>No users found for your search!</p>
-}
+
+<div class="main-container">
+  @if (items()[0]) {
+    <table>
+      <thead>
+        <tr>
+          <th>Username</th>
+        </tr>
+      </thead>
+      <tbody>
+        @for (user of items(); track user.id) {
+          <tr>
+            <td>
+              <a [routerLink]="['/users', user.username]" class="username-link">
+                {{ user.username }}
+              </a>
+            </td>
+          </tr>
+        }
+      </tbody>
+    </table>
+    <div>
+      <div scope="row" colspan="2">
+        <p>
+          {{
+            loadingService.isLoadingId(loadingKey) ? "?-? of ?" : paginationText
+          }}
+          users.
+        </p>
+      </div>
```

Comment:
We dont need the wrapping div around the <p> block, <p> is already a full width element. Also scope and colspan does not apply to div.